indico/modules/events/registration/util.py

```diff
@@ -1168,3 +1168,47 @@ class CustomTicketCode:
     def lookup_registration(cls, data: str) -> Registration | None:
         """Lookup a registration based on custom ticket code data."""
         raise NotImplementedError
+
+
+def prepare_participant_list_data(reglist, display, static_items_ids):
+    primary_headers = ['ID', 'Name']
+    dynamic_headers = [item.title for item in display]
+
+    rows = []
+    for registration in reglist:
+        row = {
+            'ID': f'#{registration.friendly_id}',
+            'Name': registration.display_full_name
+        }
+        reg_data_by_field = registration.data_by_field
+
+        for item in display:
+            reg_data = reg_data_by_field.get(item.id)
+            cell_data = None
+            if reg_data:
+                cell_data = item.field_impl.render_reglist_column(reg_data).text_value or '-'
+            row[item.title] = cell_data
+
+        for item_id in static_items_ids:
+            match item_id:
+                case 'reg_date':
+                    cell_data = format_datetime(registration.submitted_dt)
+                case 'state':
+                    cell_data = registration.state.title
+                case 'price':
+                    cell_data = registration.render_price()
+                case 'checked_in':
+                    cell_data = 'Yes' if registration.checked_in else 'No'
+                case 'checked_in_date':
+                    if registration.checked_in_dt:
+                        cell_data = format_datetime(registration.checked_in_dt)
+                case 'tags_present':
+                    if registration.tags:
+                        cell_data = ', '.join(sorted(t.title for t in registration.tags))
```

Comment:
```suggestion
                case 'checked_in_date' if registration.checked_in_dt:
                    cell_data = format_datetime(registration.checked_in_dt)
                case 'tags_present' if registration.tags:
                    cell_data = ', '.join(sorted(t.title for t in registration.tags))
```

otherwise `cell_data` remains defined in case of no checkin / tags